### DISCUSSION TITLE
More backends support

### DISCUSSION THREAD

**Discussion by hipudding** [mlfoundations/open_clip#796] [General]
I'm wondering whether it's welcome to contribute for more backends?

**Comment by ludwigschmidt** [comment]
Hi! Thanks for your interest. What do you have in mind?

**Reply by hipudding** [reply]
Can I try to add [Ascend NPU](https://www.hiascend.com/en/) as a new backend?

**Reply by ludwigschmidt** [reply]
It depends on how much that would complicate the codebase. We could look at a PR?

**Reply by hipudding** [reply]
Thanks, But I didn't start yet, I will create this PR after I have done.
